pkg/roachprod/vm/vm.go

```diff
@@ -43,6 +43,10 @@ const (
 	ArchFIPS    = CPUArch("fips")
 	ArchUnknown = CPUArch("unknown")
 
+	// IncompatibleArchRequestError is returned when requested architecture
+	// is different than the architecture discovered from the machine-type
+	IncompatibleArchRequestError = `%s-machine-type %s arch (%s) incompatible with requested arch (%s)`
```

Comment:
tiny nit: This could be a function instead so it returns an error instead of needing to do the casting yourself.

something like: 
```
func IncompatibleArchRequestError(
	provider, requestedMachineType, discoveredArch, requestedArch string,
) error {
	return errors.Newf(`%s-machine-type %s arch (%s) incompatible with requested arch (%s)`,
		provider, requestedMachineType, discoveredArch, requestedArch,
	)
}
```